pkg/cosign/tuf/client_test.go

```diff
@@ -132,6 +138,68 @@ func TestCache(t *testing.T) {
 	checkTargetsAndMeta(t, tuf)
 }
 
+func TestCustomRoot(t *testing.T) {
+	ctx := context.Background()
+	// Create a remote repository.
+	td := t.TempDir()
+	remote, r := newTufRepo(t, td, "foo")
+
+	// Serve remote repository.
+	s := httptest.NewServer(http.FileServer(http.Dir(filepath.Join(td, "repository"))))
+	defer s.Close()
+
+	// Initialize with custom root.
+	tufRoot := t.TempDir()
+	t.Setenv("TUF_ROOT", tufRoot)
+	meta, err := remote.GetMeta()
+	if err != nil {
+		t.Error(err)
+	}
+	rootBytes, ok := meta["root.json"]
+	if !ok {
+		t.Error(err)
+	}
+	if err := Initialize(ctx, s.URL, rootBytes); err != nil {
+		t.Error(err)
+	}
+	if l := dirLen(t, tufRoot); l == 0 {
+		t.Errorf("expected filesystem writes, got %d entries", l)
+	}
+
+	// Successfully get target.
+	tufObj, err := NewFromEnv(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if b, err := tufObj.GetTarget("foo.txt"); err != nil || !bytes.Equal(b, []byte("foo")) {
+		t.Fatal(err)
+	}
+	tufObj.Close()
+
+	// Force expiration on the first timestamp and internal go-tuf verification.
+	expCleanup := forceExpirationVersion(1, true)
+	if _, err = NewFromEnv(ctx); err == nil {
+		t.Errorf("expected expired timestamp from the remote")
+	}
+	expCleanup()
+
+	// Update remote targets, issue a timestamp v2.
+	updateTufRepo(t, td, r, "foo1")
+
+	// Force expiration on the first timestamp.
+	expCleanup = forceExpirationVersion(1, false)
```

Comment:
To clarify, why do we need to force the expiration again for version 1, after it's been done on line 180?

Reply:
There's two places where we're forcing verification to fail -- once in cosign when we check specifically for the timestamp, and once internally in go-tuf. The first call disables both so that internal go-tuf actually fails when calling `Update()` (otherwise `NewFromEnv` will enventually pass -- fails on our check, then passes with the Update). 

In the second call, we want to make cosign think it needs to fetch an update because the timestamp v1 is expired, but let Update fetch all the latest data from remote without forcing failure.

AS I WRITE THIS: I realize I should just split this up to to only cleanup the internal go-tuf force failure. Thanks :)